integration tests not working.

I am trying to use ``crio`` on ``ubuntu 16.04``.

I tried to run integration tests ``make integration`` and ``make localintegration``, Both fails with error 

```
 manifest contains multiple images 
```

- localintegration log
```
$ make localintegration  
rm -f "/home/kunal/go/.gopathok"
rm -rf _output
rm -f docs/*.1 docs/*.5 docs/*.8
rm -fr test/testdata/redis-image
find . -name \*~ -delete
find . -name \#\* -delete
rm -f crioctl crio kpod
make -C conmon clean
make[1]: Entering directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/conmon'
rm -f conmon.o cmsg.o conmon
make[1]: Leaving directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/conmon'
make -C pause clean
make[1]: Entering directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/pause'
rm -f pause.o pause
make[1]: Leaving directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/pause'
rm -f test/bin2img/bin2img
rm -f test/copyimg/copyimg
rm -f test/checkseccomp/checkseccomp
touch "/home/kunal/go/.gopathok"
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505275498 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccom
p  libdm_no_deferred_remove" -o crio github.com/kubernetes-incubator/cri-o/cmd/crio
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505275498 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccom
p  libdm_no_deferred_remove" -o crioctl github.com/kubernetes-incubator/cri-o/cmd/crioctl
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505275498 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccom
p  libdm_no_deferred_remove" -o kpod github.com/kubernetes-incubator/cri-o/cmd/kpod
make -C conmon
make[1]: Entering directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/conmon'
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include   -c -o conmon.o conmon.c
conmon.c: In function ‘main’:
conmon.c:1107:3: warning: ignoring return value of ‘read’, declared with attribute warn_unused_result [-Wunused-result]
   read(start_pipe_fd, buf, BUF_SIZE);
   ^
cc -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include   -c -o cmsg.o cmsg.c
cc -o conmon conmon.o cmsg.o -std=c99 -Os -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lglib-2.0
make[1]: Leaving directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/conmon'
make -C pause
make[1]: Entering directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/pause'
cc -std=c99 -Os -Wall -Wextra -static   -c -o pause.o pause.c
cc -o pause pause.o -std=c99 -Os -Wall -Wextra -static 
strip pause
make[1]: Leaving directory '/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/pause'
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505275498 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccomp  libdm_no_deferred_remove" -o test/bin2img/bin2img github.com/kubernetes-incubator/cri-o/test/bin2img
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505275498 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccomp  libdm_no_deferred_remove" -o test/copyimg/copyimg github.com/kubernetes-incubator/cri-o/test/copyimg
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505275498 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccomp  libdm_no_deferred_remove" -o test/checkseccomp/checkseccomp github.com/kubernetes-incubator/cri-o/test/checkseccomp
./test/test_runner.sh 
ERRO[0004] error copying docker://busybox to dir:/home/kunal/go/src/github.com/kubernetes-incubator/cri-o/.artifacts/busybox-image: can not copy docker://busybox:latest: manifest contains multiple images 
Error pulling docker://busybox
Makefile:119: recipe for target 'localintegration' failed
make: *** [localintegration] Error 1

```

- make integration log
```
$ make integration
docker build -t crio_dev:master .
Sending build context to Docker daemon  307.4MB
Step 1/20 : FROM golang:1.8
 ---> c0ccf5f2c036
Step 2/20 : RUN echo 'deb http://httpredir.debian.org/debian jessie-backports main' > /etc/apt/sources.list.d/backports.list
 ---> Running in 1c806a6a23d6
 ---> 2a67c71cf6e6
.
. <snip>
.
/etc/containers/registries.d/registry.access.redhat.com.yaml
 ---> 287c55f2234a
Step 20/22 : WORKDIR /go/src/github.com/kubernetes-incubator/cri-o
 ---> 9ba586de00cc
Removing intermediate container 40dae74c6885
Step 21/22 : ADD . /go/src/github.com/kubernetes-incubator/cri-o
 ---> d4a844d5d269
Step 22/22 : RUN make test/copyimg/copyimg      && mkdir -p .artifacts/redis-image      && ./test/copyimg/copyimg --import-from=docker://redis --export-to=dir:.artifacts/redis-image --signatu
re-policy ./test/policy.json
 ---> Running in 1684165c5ba3
touch "/go/.gopathok"
go build -ldflags '-X main.gitCommit="ae5739c3de0821f2282178aa9d41bf0a42afc221-dirty" -X main.buildInfo=1505271429 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux secco$
p  " -o test/copyimg/copyimg github.com/kubernetes-incubator/cri-o/test/copyimg
time="2017-09-13T02:58:04Z" level=error msg="error copying docker://redis to dir:.artifacts/redis-image: can not copy docker://redis:latest: manifest contains multiple images" 
The command '/bin/sh -c make test/copyimg/copyimg       && mkdir -p .artifacts/redis-image    && ./test/copyimg/copyimg --import-from=docker://redis --export-to=dir:.artifacts/redis-image --s
ignature-policy ./test/policy.json' returned a non-zero code: 1
Makefile:110: recipe for target 'crioimage' failed
make: *** [crioimage] Error 1

```

--- a/libpod/images/copy_ref.go
+++ b/libpod/images/copy_ref.go
@@ -59,7 +59,7 @@ type copySource struct {
 
 // NewImage creates a new image from the given system context
 func (c *CopyRef) NewImage(sc *types.SystemContext) (types.Image, error) {
-	src, err := c.NewImageSource(sc, nil)
+	src, err := c.NewImageSource(sc)
 	if err != nil {
 		return nil, err
 	}
@@ -82,10 +82,9 @@ func selectManifestType(preferred string, acceptable, supported []string) string
 }
 
 // NewImageSource creates a new image source from the given system context and manifest
-func (c *CopyRef) NewImageSource(sc *types.SystemContext, manifestTypes []string) (src types.ImageSource, err error) {
+func (c *CopyRef) NewImageSource(sc *types.SystemContext) (src types.ImageSource, err error) {
 	// Decide which type of manifest and configuration output we're going to provide.
-	supportedManifestTypes := []string{v1.MediaTypeImageManifest, docker.V2S2MediaTypeManifest}
-	manifestType := selectManifestType(c.preferredManifestType, manifestTypes, supportedManifestTypes)
+	manifestType := selectManifestType(c.preferredManifestType, nil, nil)
 	// If it's not a format we support, return an error.
 	if manifestType != v1.MediaTypeImageManifest && manifestType != docker.V2S2MediaTypeManifest {
 		return nil, errors.Errorf("no supported manifest types (attempted to use %q, only know %q and %q)",

--- a/pkg/storage/image.go
+++ b/pkg/storage/image.go
@@ -169,7 +169,7 @@ func (svc *imageService) CanPull(imageName string, options *copy.Options) (bool,
 	if err != nil {
 		return false, err
 	}
-	rawSource, err := srcRef.NewImageSource(options.SourceCtx, nil)
+	rawSource, err := srcRef.NewImageSource(options.SourceCtx)
 	if err != nil {
 		return false, err
 	}

--- a/vendor.conf
+++ b/vendor.conf
@@ -5,7 +5,7 @@ k8s.io/apimachinery release-1.7 https://github.com/kubernetes/apimachinery
 k8s.io/apiserver release-1.7 https://github.com/kubernetes/apiserver
 #
 github.com/sirupsen/logrus v1.0.0
-github.com/containers/image abb4cd79e3427bb2b02a5930814ef2ad19983c24
+github.com/containers/image d17474f39dae1da15ab9ae033d57ebefcf62f77a
 github.com/docker/docker-credential-helpers d68f9aeca33f5fd3f08eeae5e9d175edf4e731d1
 github.com/ostreedev/ostree-go master
 github.com/containers/storage f8cff0727cf0802f0752ca58d2c05ec5270a47d5

